### DISCUSSION TITLE
How to restore an accidentally deleted Collection

### DISCUSSION THREAD

**Discussion by TheHolyWaffle** [Kong/insomnia#6026] [Help] [answered]
So, I was too fast in cleaning up some of the Collections that I had sitting around. But I accidentally deleted a very valuable collection. Unfortunately, I didn't have sync enabled. 

Is there any way to restore this Collection?

**Comment by geoshak** [comment]
Yeap, it's FREAKING absurd.
I was cleaning up, and the app showed a valuable collection as 'New Document', which was the name for another collection with nothing in it. I deleted it, and it deleted the valuable collection instead.

I FREAKING HATE THIS. Got me so mad.
Used to love this app.

**Reply by cmarqs** [reply]
The exactly same issue happened to me: I was deleting "New Document" and my whole collection gone. I've remove these lines from insomnia.Workspace.db...
```
{"$$deleted":true,"_id":"wrk_1ad198a9eb5747bfadce269f8a88f41f"}
{"$$deleted":true,"_id":"wrk_510bf395bd7f482c98a02b68526920a7"}
```
...to try restore it, but the "New Document" and "My Collection" re-appeared but no endpoints inside. Waiting something on this issue now. :(

Edit: I was saved by Time Machine from Apple. I've restored all files and the collection showed up.

**Comment by klit79** [comment, marked as answer]
It is because auto sort base on time it is opened.

For me, I open the collection to check if there is anything valuable before delete.
Then I delete immediately without checking the name because it is the same name anyway.
However, by the time I clicked back, it is sorted base on time it is opened, so the one I want to delete is not deleted and the one I do not want to delete is deleted.

I managed to recover the collection 6 months ago by running this on Mac

cd /Users/user/Library/Application Support/Insomnia
cp backups/8.5.1/*.db .

**Reply by JoshukLite** [reply]
@klit79 agree. I just had the same issue and such UX is very not intuitive. I did not expect a collection to change its position and I removed it accidentally. Now I don't have any backups except some very old ones I made a year ago and I need to recreate the whole collection again.
It's a pity that it's so easy to remove a whole collection or a request without the ability to undo the action taking into account the not user-friendly UX.

**Reply by richardw33d** [reply]
You're just my hero! Thanks

**Reply by Kerubi-5** [reply]
I hope they fix this soon, this comment saved me hours of head-ache and work 💯

**Reply by giovannibenussi** [reply]
Same for me. I clicked on a collection, it was empty, so I proceeded to delete it, and then it changed its position and I deleted an important collection.
Thanks for sharing the command to recover the collection!
